gems/aws-sdk-core/spec/aws-sdk-core/credentials_provider_chain_spec.rb

```diff
@@ -0,0 +1,141 @@
+            #   assume_role_provider_class
+            # )
+            # stub_const(
+            #   'AWS::SDK::STS::AssumeRoleWebIdentityCredentialsProvider',
+            #   assume_role_web_identity_provider_class
+            # )
+            assume_role_provider = double(
+              'AssumeRoleCredentialsProvider',
+              identity: credentials(values)
+            )
+            assume_role_web_identity_provider = double(
+              'AssumeRoleWebIdentityCredentialsProvider',
+              identity: credentials(values)
+            )
+            # stub_const(
+            #   'AWS::SDK::STS::AssumeRoleCredentialsProvider',
+            #   assume_role_provider_class
+            # )
+            # stub_const(
+            #   'AWS::SDK::STS::AssumeRoleWebIdentityCredentialsProvider',
+            #   assume_role_web_identity_provider_class
+            # )
+            sts = AWS::SDK::STS
+            allow_any_instance_of(sts::AssumeRoleCredentialsProvider)
+              .to receive(:new).and_return(assume_role_provider)
+            allow_any_instance_of(sts::AssumeRoleWebIdentityCredentialsProvider)
+              .to receive(:new).and_return(assume_role_web_identity_provider)
+          when 'ssoResponse'
+            role_provider = double(
+              'RoleCredentialsProvider',
+              identity: credentials(values)
+            )
+            stub_const('AWS::SDK::SSO::RoleCredentialsProvider', Class.new)
+            sso = AWS::SDK::SSO
+            allow_any_instance_of(sso::RoleCredentialsProvider)
+              .to receive(:new).and_return(role_provider)
+          when 'processResponse'
+            process_provider = double(
+              'ProcessCredentialsProvider',
+              identity: credentials(values)
+            )
+            allow_any_instance_of(ProcessCredentialsProvider)
+              .to receive(:new).and_return(process_provider)
+          when 'containerResponse'
+            container_provider = double(
+              'ContainerCredentialsProvider',
+              identity: credentials(values)
+            )
+            allow_any_instance_of(ContainerCredentialsProvider)
+              .to receive(:new).and_return(container_provider)
+          when 'instanceResponse'
+            instance_provider = double(
+              'InstanceCredentialsProvider',
+              identity: credentials(values)
+            )
+            allow_any_instance_of(InstanceCredentialsProvider)
+              .to receive(:new).and_return(instance_provider)
+          else
+            raise 'Unknown input type'
+          end
+        end
+      end
+
+      test_cases.each do |test_case|
+        skippable = [
+          'JVM system properties: credentials',
+          'Boto 2 config file',
+          'Windows AWS SDK store'
+        ]
+
+        it "passes: #{test_case['documentation']}" do
+          skip if skippable.include?(test_case['documentation'])
```

Comment:
Is documentation validated or just free form?  It seems a little risky to rely on human entered descriptions for whether we should skip or not

Reply:
This was the only indicator I had. Can probably come up with something else in the json